### PR TITLE
rework to better assert fields with multiple types

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew

### CODE REVIEW

File: dart/tool/dart/generate_dart.dart [hunk 1]
```diff
@@ -869,6 +864,26 @@ class Type extends Member {
             gen.writeln(
                 '// assert obj.${field.generatableName} is ${type.name}');
           }
+        } else if (type.isMultipleReturns) {
+          //
```
Comment (devoncarew) [inline dart/tool/dart/generate_dart.dart]: Did you mean to keep these comments?

**Comment by devoncarew** [comment]
lgtm

File: dart/tool/dart/generate_dart.dart (hunk 1 above)
Comment (danrubel) [inline dart/tool/dart/generate_dart.dart]: Nope. Good catch! Removed.

**Comment by danrubel** [comment]
Merged https://github.com/dart-lang/vm_service_drivers/commit/6300c13904084c21d35d60cf36b7b0713859a1ee

**Pull request closed without merging**